MetricsUploader/include/MetricsUploader/MetricsUploader.h

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_METRICS_UPLOADER_METRICS_UPLOADER_H_
+#define ORBIT_METRICS_UPLOADER_METRICS_UPLOADER_H_
+
+#ifdef _WIN32
+#include <windows.h>
+#endif  // WIN32
+
+#include <absl/synchronization/mutex.h>
+
+#include <cstdint>
+#include <memory>
+#include <string>
+#include <string_view>
+
+#include "MetricsUploader/Result.h"
+#include "OrbitBase/Result.h"
+#include "orbit_log_event.pb.h"
+
+namespace orbit_metrics_uploader {
+
+constexpr std::string_view kMetricsUploaderClientDllName = "metrics_uploader_client";
+
+// This class is used for sending log events from Orbit. It works only on Windows and only if
+// metrics_uploader_client.dll is available. The types of logs that could be send are defined in
+// orbit_client_protos::OrbitLogEvent::LogEventType.
+//
+// Only one instance of the class is allowed.
+//
+// Usage example:
+//
+// auto metrics_uploader = MetricsUploader::CreateMetricsUploader();
+// if (metrics_uploader.has_value()) {
+//   metrics_uploader.value()->SendLogEvent(...);
+// }
+//
+class MetricsUploader {
+ public:
+  // Create a MetricsUploader instance, load metrics uploader client library if available and starts
+  // metrics uploader client when called first time. Return the instance if there are no errors and
+  // ErrorMessage otherwise.
+  [[nodiscard]] static ErrorMessageOr<std::unique_ptr<MetricsUploader>> CreateMetricsUploader(
+      std::string_view client_name = kMetricsUploaderClientDllName);
+
+  // Unload metrics_uploader_client.dll
+  ~MetricsUploader();
+
+  MetricsUploader(const MetricsUploader& other) = delete;
+  MetricsUploader(MetricsUploader&& other) = delete;
```

Comment:
With the new way of handling `FreeLibrary` a move constructor/assignment operator can be easily defined:

```
MetricsUploader(MetricsUploader&& other) : 
  send_log_event_addr_{other.send_log_event_addr_},
  client_name{std::move(client_name)}
{
  other.send_log_event_addr_ = nullptr;
}
```

The move assignment-operator is roughly the same, but needs to check for self assignment:
```
MetricsUploader& MetricsUploader::operator= (MetricsUploader&& other) {
  if(&other == this) return *this;

  send_log_event_addr_ = other.send_log_event_addr_;
  other.send_log_event_addr_ = nullptr;

  client_name = std::move(other.client_name);
  return *this;
}
```

And the destructor may only call `FreeLibrary` when `send_log_event_addr_` is `nullptr`:

```
MetricsUploader::~MetricsUploader() {
  if (send_log_event_addr_ == nullptr)
    return;

  // destructor work....
}
```

Reply:
You would also need to move `metrics_uploader_client_dll_` (which I suggested to be a member variable in a comment above)

Reply:
Added move constructor/assignment for the MetricsUploader for Windows